### ISSUE TITLE
Audio Pops in Timeline and Export

### ISSUE THREAD

**Issue by snr8065** [OpenShot/openshot-qt#1186]
**System Details:** 
* Operating System / Distro: OSX 10.10.5 Yosemite
* OpenShot Version: 2.4.1

My audio track sounds clear in other programs and in OpenShot's Preview Audio mode. But when played via the OpenShot Timeline AND after Exporting, there are intermittent pops and crackles. I've tried 24bit WAV and 192kbps m4a - both 44.1k sample rates. 

At this point, I don't want to redo my video edits, so I'm going to export a silent version and lay in the audio with a different program. Like I said, it plays clean everywhere else except Timeline and Export, so it's for sure something within the OpenShot architecture.

I just wanted to let you know the audio bug still exists in the latest version. Best of luck, devs.

**Comment by osve001** [comment]
Moving my comments from this thread:
#1069 Audio popping & crackling in exported video with overlaid audio tracks.
I think it it fits better here.
-----
Windows 7
Openshot 2.4.1

I'm afraid I'm suffering the same problem. The video recording is perfect (no crackling at all). No other audio track but the one in the original video (video game recording with comments on the go from Android tablet). When playing back the project, the sound crackles and, what's worse, the crackling remains in the exported video. I think that the problem begins when importing the video.
Any ideas? Thanks in advance.

-----

I've tried Windows 10 and Ubuntu 16.04. The same problem. Then, I extracted the audio (soundtrack) from the video (that's to convert it to mp3) and added it to the Openshot project. I muted the video and ... It works fine, the audio is perfect (no crackling, as in the original recording).
So my conclusion is that there could be a problem with the audio when importing the video.
I hope this helps.
Thank you.

**Comment by osve001** [comment]
I can tell you that the audio also crackles with Windows Movie Maker 2012 (Build 16.4.3xxx).
Thank you

**Comment by DylanC** [comment]
@osve001 - Maybe a problem with your system configuration or the audio file?

**Comment by DylanC** [comment]
@snr8065 - I think a fix has been added to sort audio popping. Can you try the latest daily build?

**Comment by osve001** [comment]
I'm working with OpenShot Version 2.4.1 (on Windows 7, Windows 10 and Ubuntu 16.04)
Is there a newer one?
Thanks

**Comment by osve001** [comment]
I'm working with OpenShot Version 2.4.1 (on 1 PC: Windows 7 - Ubuntu 16.04  and another laptop: Windows 10)
Is there a newer one?
With these tests I would rule out any hw flaws or wrong settings. And what's more, and I mentioned it, if I import the audio (previously extracted) and mute the video, everything is fine. So, no problem when importing audio. The issue with the soundtrack appears when importing the video.
I hope this helps.
Thanks

**Comment by DylanC** [comment]
@osve001 - There are newer versions with the latest fixes called the daily builds. I would recommend you try one out in case the audio fix for popping works for you.

**Comment by fabrixx** [comment]
I've been trying to solve this problem for months. Today I found an experimental solution subdividing the video clips that were on a single track on three different tracks leaving the audio track unaltered as in the picture. the sound popping has disappeared both from the preview and from the exported file. I hope he can help someone.
![aaaschermata](https://user-images.githubusercontent.com/1190942/39950406-1363251c-5581-11e8-962e-b15d66ce09ee.png)

**Comment by N3WWN** [comment]
Is the audio popping occurring only where the video tracks butt up against each other?

**Comment by fabrixx** [comment]
Yes. Between two clips. I noticed also with single 1audio+1single video tracks  if i leave an empty space from first to second clip, pop move from second to third or last clip.  So strange.

**Comment by N3WWN** [comment]
This may be an artifact that also causes #1337 and #1478 .

If your audio is always on a different track, try silencing each of the video-only clips (`Right-Click` on Clip -> `Volume` -> `Entire Clip` -> `Level 0%`) and see if the audio issue is still there.  

I'm guess that, if you mute the video clips, that it won't matter where they are in the timeline or what track they are on...

**Comment by fabrixx** [comment]
I've already tried to silence the video tracks but I still have the problem. I have no problems with the video but only on the audio. I also tried with wav liblamemp3 AAC with various bitrates. I've only encountered the problem for a few months, before everything was ok. I have the same problem on Notebook & Desktop (Arch Linux). I have tested Openshot latest snapshot and stable.

**Comment by fabrixx** [comment]
UPDATE: Version 2.4.2-rc1 for Linux greatly improves stability and usability and produces a much more purring buzz that tends to fade immediately after each transition.

**Comment by fabrixx** [comment]
Today I did some tests and I can give more precise details about my solution of movement on multiple tracks.
I only exported the video and then added the tracks to the mp4 file without video transitions.
The problem is due to the audio transitions between the tracks. If I have two audio clip A clip and B clip on the same track to which an audio transition is applied I have the annoying noise both in preview and in export.
The problem is solved if I move Clip B to the highest track as in the image.
![soluz1](https://user-images.githubusercontent.com/1190942/47269144-29365680-d55a-11e8-9fda-496cc657c5b1.jpg)
![soluz1a](https://user-images.githubusercontent.com/1190942/47269145-29ceed00-d55a-11e8-949e-0a4be29d52c8.jpg)

**Comment by speedytechdev** [comment]
@snr8065 or @fabrixx Is this issue still happening?

**Comment by fabrixx** [comment]
Yes. I edited some video for my last trip and have this bad sound in every video:
https://www.youtube.com/playlist?list=PLxp9ZhyOHJury7jTXFzYvpolgkfIYGXqq

Solution for me is export video and after re add audio track.

**Comment by speedytechdev** [comment]
We will work as fast as possible to fix the issue, we are just low on volunteer staff. Sorry for any inconvenience this has caused you. If you have any questions please feel free to ask us.

**Comment by kristopolous** [comment]
i'm totally experiencing this as well ... the format for openshot used to be a rather parsable xml but it looks to be a large json now. I was thinking there'd be a rather trivial way to parse the json and then construct a shell script to use ffmpeg and relayer the audio ... but this doesn't look readily achievable.

It looks like that no audio transition is done at all if you open things up in audacity ... the two audios are simply layered on top of each other at 100% volume and then one goes from 100%->0% instantly which is what's causing the clipping. 

It's really too bad as the older 1.x version of the software didn't have this issue.

**Comment by Polda18** [comment]
I moved to Shotcut that also has the performance issues with FHD video playback, but it doesn't seem to have the audio crackling in the exported video issue. I can easily just lower the resolution of the video for preview and return it back to FHD when it's ready for export. Nothing fancy, but it works. Look at this testing clip I made as preparation for my gaming channel:
https://www.youtube.com/watch?v=v5zv2n2gvlQ
